### PR TITLE
Move cfp and voting to date config

### PR DESCRIPTION
**Pull request by anativ**
This branch has 2 features:

1. auto close/open voting/cfp on 'config date' - no need to upload version at 00:00
1. block CFP (server side) + block voting (server side)

### CODE REVIEW

**Comment by rantav** [comment]
it's probably better that @amitzur reviewed this, since it isn't urgent. 

Just a question, what timezone did you use?

**Comment by anativ** [comment]
Israel - ```moment().tz('Israel')```

File: server/db/controllers/proposals.js [hunk 1]
```diff
@@ -47,8 +47,8 @@ async function get(req, res) {
 /**
  * Add a Proposal
  */
-export function add(req, res) {
-  if (!req.user) return res.sendStatus(401);
+export function add(req, res) {  
+  if (!req.user || !eventConfig.cfp) return res.sendStatus(401);
```
Comment (rantav) [inline server/db/controllers/proposals.js]: so add is also update, right? 
If so, then you need to allow `isReversimTeamMember` here event if `eventConfig.cfp` is false

File: server/init/eventConfig.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import moment from 'moment-timezone';
+
+const config = {
+  cfpStateDate: "2018-04-25",
```
Comment (rantav) [inline server/init/eventConfig.js]: typo: `cfpStateDate` (and in line 10)

File: server/init/eventConfig.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  cfpEndDate: "2018-05-31",
+  votingStartDate: "2018-06-01",
+  votingEndDate: "2018-06-10",
+}
+
+const cfp = moment().tz('Israel').isBetween(config.cfpStateDate, config.cfpEndDate, 'day', "[]");
+const voting = moment().tz('Israel').isBetween(config.votingStartDate, config.votingEndDate, 'day', "[]");
```
Comment (rantav) [inline server/init/eventConfig.js]: so these two fields get initialized when the server starts, right? 
so they won't get updated over time unless the container is restarted, which is something you don't control in heroku as far as I know (and anyway it doesn't seem to be a good thing to rely on)

File: server/init/routes.js [hunk 4]
```diff
@@ -83,6 +84,7 @@ export default (app) => {
       team: team.map(user => transformUser(user, req.user)),
       messages,
       sponsors,
+      eventConfig,
```
Comment (rantav) [inline server/init/routes.js]: huh, ok so there's just one api endpoint which returns everything at once? 
oh well, ok...

File: server/db/controllers/proposals.js (hunk 1 above)
Comment (anativ) [reply server/db/controllers/proposals.js]: no, there is an update method

```javascript
/**
 * Update a proposal
 */
async function update(req, res) {
...
```

File: server/init/eventConfig.js (hunk 3 above)
Comment (anativ) [reply server/init/eventConfig.js]: good catch

File: server/init/routes.js (hunk 4 above)
Comment (anativ) [reply server/init/routes.js]: Yes, this is not optimal, but I didn't want to change it in this PR

File: server/init/eventConfig.js (hunk 2 above)
Comment (anativ) [reply server/init/eventConfig.js]: 😲

File: client/src/components/Hero.js [hunk 5]
```diff
@@ -5,13 +5,16 @@ import Messages from './Messages';
 import { Button, Container } from 'reactstrap';
 import { REVERSIM_SUMMIT } from '../utils';
 import { Link } from 'react-router-dom';
-import { cfp, voting } from '../features';
-
 
 class Hero extends Component {
 
+  static defaultProps = {
+    eventConfig: {},
```
[the remark this answers is not in the document]
Comment (anativ) [reply client/src/components/Hero.js]: No validation at all in *all the codebase*

**Comment by anativ** [comment]
@rantav ready for round 2 👊

**Pull request merged**